### PR TITLE
Properly support dependency on a java-library module

### LINKED ISSUE #7729: Quarkus 1.3.0.CR2: Dependency of gradle library submodule not included
**Describe the bug**
I have a gradle project with multiple modules. The library module (gradle plugin `java-library`) contains some common code shared between several modules.
The library module also includes a dependency (e.g. `io.quarkus:quarkus-kubernetes-client`) as `api` dependency. Therefore it should be included as transitive dependency in the application module including the library module. 

1. When running `./gradlew :application:quarkusDev`, this fails with an `UnsatisfiedResolutionException` for the `KubernetesClient` in 1.3.0.CR2 but works in Quarkus 1.2.1.Final.

2. When running `./gradlew :application:quarkusBuild` the first time, the `UnsatisfiedResolutionException` for my `BeanWithInjection` class of the library project.

3. When running first `./gradlew :application:quarkusDev`, canceling it and then running `./gradlew :application:quarkusBuild`, I get the first situation again (`UnsatisfiedResolutionException` for the `KubernetesClient`).

I attached a reproducer below. 

**Expected behavior**
The api dependency of the library module is included as transitive dependency in the module including the library.

**Actual behavior**
The transitive dependency is not included any more.

**To Reproduce**
1. Download reproducer project [multi-module-transitive-not-working.zip](https://github.com/quarkusio/quarkus/files/4312438/multi-module-transitive-not-working.zip) and unzip it.
2. In the reproducer project, run `./gradlew :application:quarkusDev` to get the exception: `javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.fabric8.kubernetes.client.KubernetesClient`
3. Run `./gradlew clean :application:quarkusBuild` to get the 2nd exception: `javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type org.acme.BeanWithInjection`
4. Run `./gradlew :application:quarkusDev`, cancel it after it failed to build and then run `./gradlew :application:quarkusBuild` to get the same error as in step 2. 

**Environment (please complete the following information):**
 - Windows 10
 - gradle 6.2
 - quarkus 1.3.0.CR2

### PR DESCRIPTION
**Pull request by aloubyansky**
And collect first-level extension dependencies in the complete dependency tree instead of project direct extension dependencies.

Fixes #7729

### CODE REVIEW

File: devtools/gradle/src/main/java/io/quarkus/gradle/AppModelGradleResolver.java [hunk 1]
```diff
@@ -306,7 +337,7 @@ static AppDependency toAppDependency(ResolvedArtifact a) {
         return new AppDependency(appArtifact, "runtime");
     }
 
-    private Dependency processQuarkusDir(ResolvedArtifact a, Path quarkusDir, AppModel.Builder appBuilder) {
+    private Dependency processQuarkusDir(Object a, Path quarkusDir, AppModel.Builder appBuilder) {
```
Comment (gastaldi) [inline devtools/gradle/src/main/java/io/quarkus/gradle/AppModelGradleResolver.java]: Maybe remove this first parameter?

Comment (aloubyansky) [reply devtools/gradle/src/main/java/io/quarkus/gradle/AppModelGradleResolver.java]: oh, that's a left-over, sure. Thanks.

File: devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java [hunk 2]
```diff
@@ -142,16 +143,36 @@ private void afterEvaluate(Project project) {
     }
 
     private void configProjectDependency(Project project, Project dep) {
+        dep.afterEvaluate(p -> {
+            setupTaskDependencies(project, p);
+            final Map<String, Project> childProjects = dep.getChildProjects();
+            if (!childProjects.isEmpty()) {
```
Comment (gastaldi) [inline devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java]: This `if` is not necessary

**Review by gastaldi — APPROVED** [review]
LGTM

**Pull request merged**